### DISCUSSION TITLE
Implementing UVTools into PrusaSlicer

### DISCUSSION THREAD

**Discussion by NickSica** [sn4k3/UVtools#699] [General]
Hey, I'm currently working on the CTBv4 integration into PrusaSlicer and I've used info from this repo as a jumping off point to help me. I just wanted to double check that you were fine with it. I also liked some of your other ideas and tools and was wondering if you would be fine with me implementing some of them into PrusaSlicer as well once I'm done with the CTB file implementation.

**Comment by sn4k3** [comment]
UVtools license (AGPL) is compatible and same as with PrusaSlicer (AGPL). And so, re-use is allowed as long code is make public and open, which is. Feel free to grab whatever you want.
Also note that we do not own the file specs / structure, they are just sequences of bytes, so we could never make it hostage anyway ;)

**Reply by NickSica** [reply]
Sounds great! It was more basing implementation of things like suction cup detection off of your implementation than any filetypes I end up implementing. Your work has also been a huge help in decoding ctb files.

**Reply by sn4k3** [reply]
suction cup and resin trap detection should be better implemented with the mesh data. The UVtools implementation (pixel processing) is slow. Lychee for example does it at mesh level (I guess) and is much faster to almost instant
